### PR TITLE
Add documentation for methods

### PR DESCRIPTION
**Pull request by lokeshh**
Addresses #4

### CODE REVIEW

File: lib/statsample-glm/glm/base.rb [hunk 1]
```diff
@@ -28,7 +28,20 @@ def initialize ds, y, opts={}
                             .const_get("#{algorithm}").const_get("#{method}")
                             .new(@data_set, @dependent, @opts)
       end
-      
+
+      # Returns the coefficients of trained model
+      #
+      # @param [Symbol] as_a Specifies the form of output
+      #
+      # @return [Array, Hash] coefficients of the model
```
Comment (agisga) [inline lib/statsample-glm/glm/base.rb]: In this and some other methods it looks like, if `as_a=:array` is specified, then a `Daru::Vector` is returned, rather than an `Array`... Is that an issue? (If it is, I think that it can be a separate pull request though)

Comment (v0dro) [inline lib/statsample-glm/glm/base.rb]: You're right it will need to be fixed.

Comment (agisga) [inline lib/statsample-glm/glm/base.rb]: Addressed in #19

**Comment by lokeshh** [comment]
`log_likelihood` defined [here](https://github.com/SciRuby/statsample-glm/blob/e8466dc9514f520bf3e692e29202020201728294/lib/statsample-glm/glm/base.rb#L85) Could someone please explain what it does? I am little aware of the likelihood function but don't know how it's related here. Also, it doesn't take any input and returns a single value, so it's surprising me and I am not sure what is it doing.

**Comment by agisga** [comment]
@lokeshh The log-likelihood is a function of the unknown parameters (such as the regression coefficients). The MLE (maximum likelihood estimation) algorithm aims to find values for the unknown parameters, at which the log-likelihood is maximized. `log_likelihood` is the optimal value of the log-likelihood function (i.e., the value of the log-likelihood function at the MLE solution).

File: lib/statsample-glm/glm/base.rb [hunk 2]
```diff
@@ -74,6 +105,24 @@ def fitted_mean_values
         @regression.fitted_mean_values
       end
 
+      # Returns for every data point obtained using the model
```
Comment (agisga) [inline lib/statsample-glm/glm/base.rb]: I think here is a typo. Do you mean "Returns _the residual_ for (...)"?

File: lib/statsample-glm/glm/base.rb [hunk 3]
```diff
@@ -49,6 +63,23 @@ def coefficients as_a=:vector
         end
       end
 
+      # Returns the standard error obtained using the model
```
Comment (agisga) [inline lib/statsample-glm/glm/base.rb]: Rather say "Returns the standard errors for the coefficient estimates".
It looks to me like it returns a vector of standard errors, containing an individual standard error for each coefficient.

Comment (lokeshh) [inline lib/statsample-glm/glm/base.rb]: Fixed

File: lib/statsample-glm/glm/base.rb (hunk 2 above)
Comment (lokeshh) [inline lib/statsample-glm/glm/base.rb]: I updated it.

File: lib/statsample-glm/glm/base.rb [hunk 4]
```diff
@@ -74,14 +105,55 @@ def fitted_mean_values
...
+      #
+      # @example
+      #   require 'statsample-glm'
+      #   data_set = Daru::DataFrame.from_csv "spec/data/logistic.csv"
+      #   glm  = Statsample::GLM.compute data_set, "y", :logistic, constant: 1
+      #   glm.degree_of_freedom
+      #     # => 47
+      #
       def degree_of_freedom
         @regression.degree_of_freedom
```
Comment (lokeshh) [inline lib/statsample-glm/glm/base.rb]: I found at most of the places that it's named as "degrees of freedom" and not "degree of freedom". Should we change it?

Comment (v0dro) [inline lib/statsample-glm/glm/base.rb]: This is a question for @agisga

Comment (v0dro) [inline lib/statsample-glm/glm/base.rb]: But be sure to not break backward compatibility even if you rename the method (alias it)

Comment (agisga) [inline lib/statsample-glm/glm/base.rb]: I agree. I think, the plural "degrees" is more correct.

Similarly, maybe we should change `#standard_error` to the plural form `#standard_errors`, because it returns a vector of standard errors rather than just one value. What do you guys think?

Comment (v0dro) [inline lib/statsample-glm/glm/base.rb]: I'm ok with it. Just be sure to make the appropriate aliases to not break backwards compatibility. We'll remove the alias after 2-3 releases.

**Comment by lokeshh** [comment]
I think it can be merged now. I haven't changed the method names in this PR because it was for documentation purpose. For that I have this PR which also refactor some code https://github.com/SciRuby/statsample-glm/pull/28
